Editor/Window/GameLiftRequestAdapter.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Amazon.GameLift;
+using Amazon.GameLift.Model;
+using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Editor.Window
+{
+    public interface IAmazonGameLiftClientFactory
+    {
+        IAmazonGameLiftClientWrapper Get(string profile);
+    }
+
+    public class AmazonGameLiftClientFactory : IAmazonGameLiftClientFactory
+    {
+        private readonly CoreApi _coreApi;
+
+        public AmazonGameLiftClientFactory(CoreApi coreApi)
+        {
+            _coreApi = coreApi;
+        }
+
+        public IAmazonGameLiftClientWrapper Get(string profile)
+        {
+            var credentials = _coreApi.RetrieveAwsCredentials(profile);
+            var client = new AmazonGameLiftClient(credentials.AccessKey, credentials.SecretKey);
+            return new AmazonGameLiftWrapper(client);
+        }
+    }
+
+    public class GameLiftRequestAdapter
+    {
+        private GameLiftPlugin _gameLiftConfig;
+        private string _fleetName;
+        private string _computeName = "ComputerName-ProfileName";
+        private string _ipAddress = "120.120.120.120";
+        private string _fleetId;
+        public const string FleetLocation = "custom-location-1";
+        private const string FleetDescription = "Created By Amazon GameLift Unity Plugin";
+        private VisualElement _container;
+
+        public GameLiftRequestAdapter(GameLiftPlugin gameLiftConfig)
+        {
+            _gameLiftConfig = gameLiftConfig;
+        }
+
+        internal async Task<bool> CreateAnywhereFleet(string fleetName)
+        {
+            if (_gameLiftConfig.GameLiftWrapper != null)
+            {
+                var success = await CreateCustomLocationIfNotExists(FleetLocation);
+                if (success)
+                {
+                    var fleetId = await CreateFleet(ComputeType.ANYWHERE, FleetLocation,fleetName);
+                    var fleetNameResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.FleetName, fleetName);
+                    var fleetIdResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.FleetId, fleetId);
+                    return fleetNameResponse.Success && fleetIdResponse.Success;
```

Comment:
Since there are multiple ways that this method can fail to create a fleet, we should be using an error code enum, or throwing exceptions so that the consumer of this method can handle each case (we'll want the UI to be responsive of the problem not just that there is a problem)

Reply:
I have implemented a version of error handling within this method. Is this a suitable way of doing it? or is there a better/easier version you can think of.

Reply:
This version is fine by me